src/connectionController.ts

```diff
@@ -354,17 +335,54 @@ export default class ConnectionController {
 
     if (
       this._currentConnection &&
-      connectionToRemove === this._currentConnectionInstanceId
+      connectionId === this._currentConnectionInstanceId
     ) {
       await this.disconnect();
     }
 
-    this.removeConnectionConfig(connectionToRemove);
+    this.removeConnectionConfig(connectionId);
 
     vscode.window.showInformationMessage('MongoDB connection removed.');
     return Promise.resolve(true);
   }
 
+  public async onRemoveMongoDBConnection(): Promise<boolean> {
+    log.info('mdb.removeConnection command called');
+
+    // Ensure we aren't currently connecting or disconnecting.
+    if (this._connecting) {
```

Comment:
I'm not sure I follow. Here we're avoiding removing a connection if there's currently a connecting operation underway. The comment is actually not right though - it's not checking if there's a disconnecting operation going on. I'll fix rq thanks

Reply:
Yeah, I get what is happening here. Sorry sometimes I am not able to formulate it in a good way :) My question is if in this controller I need to check if I am already connected or not, how I can do it. For example, I want to show a different template in the playground depending on whether I am connected or not.

Reply:
Ah gotcha - I don't think we have a helper for that yet. A way to do it would be to add a method `isCurrentlyConnected` or something and then reference the variable in connection controller `_currentConnectionInstanceId` to see if that's null. When it's null it means there is no connection active. `_currentConnectionInstanceId`

Reply:
👍